### PR TITLE
New NPCs for the player to trade with

### PR DESCRIPTION
**Pull request by Brytenwalda**
**Summary**

```SUMMARY: Content "New NPCs for the player to trade with"```

#### Purpose of change
I added Free Merchant traders to gas stations and hunters to cabins and campsites to add some variety to the game - players will now have NPCs to trade and interact with. They should be rather lore-friendly and expand the lore re Free Merchants. 

#### Describe the solution
I modified mapgen files for cabins, gas stations and campsites to add new NPCs and modified npc files to put new types in game. 

#### Testing
I encountered no issues in-game and I've been using these NPCs for a while now.

### CODE REVIEW

**Comment by curstwist** [comment]
I'd recommend moving the place_npcs to a nested mapgen, then placing the nested map in the locations.  This way, you can set the NPC to a spawn chance instead of being there 100% of the time.

You can see examples of NPC nested mapgen at: https://github.com/CleverRaven/Cataclysm-DDA/blob/master/data/json/mapgen/nested/city_npc_nested_spawns.json 

You can simply put the place_npcs entry into the nest's object entry and nothing else, or, take the opportunity to also change what is spawning around it in the locations (say more storage, their base area like a bed, brazier, etc).  

the main mapgen entry gets something like this:
`"place_nested": [ { "chunks": [ [ "chef_s_restaurant", 5 ], [ "null", 95 ] ], "x": 7, "y": 18 } ]`

This has my chef spawning 5% of the time and 95% they restaurant will be empty.


Thanks a lot for adding more NPCs!

**Comment by Night-Pryanik** [comment]
> I'd recommend moving the place_npcs to a nested mapgen, then placing the nested map in the locations. This way, you can set the NPC to a spawn chance instead of being there 100% of the time.

We really need an ability to directly set chance to spawn for NPCs.

File: data/json/npcs/classes.json [hunk 1]
```diff
@@ -438,5 +438,48 @@
...
+        "level": { "mul": [ { "one_in": 3 }, { "sum": [ { "dice": [ 2, 2 ] }, { "constant": -2 }, { "one_in": 4 } ] } ] }
+      },
+      { "skill": "dodge", "bonus": { "rng": [ 4, 8 ] } }
+    ]
+  },
+  {
+    "type": "npc_class",
+    "id": "FM_TRADER",
+    "name": "FM_TRADER",
+    "job_description": "I am trading stuff",
```
Comment (Rail-Runner) [inline data/json/npcs/classes.json]:
```suggestion
    "job_description": "I am trading stuff.",
```

**Comment by LaVeyanFiend** [comment]
Why close? 😢

**Pull request closed without merging**